src/components/FilterBarInline/FilterBarInline.tsx

```diff
@@ -114,7 +106,7 @@ export const FilterBarInline: ChakraComponent<
               isBlockElement: layout === "column",
               closeOnBlur: layout === "row",
               layout: layout,
-              width: finalWidth,
+              width: layout === "row" ? "fitContent" : "full",
```

Comment:
Enforces `fitContent` with `row` layout and `full` with `column` as the component does currently. 

If `fitContent` is passed here, but the viewport is mobile, then `MultiSelect` knows to set `width` to "100%" (in `MultiSelect`'s theme file).